pkg/readiness/object_tracker.go

```diff
@@ -123,10 +128,36 @@ func (t *objectTracker) CancelExpect(o runtime.Object) {
 		return
 	}
 
-	delete(t.expect, k)
-	delete(t.seen, k)
-	delete(t.satisfied, k)
-	t.cancelled[k] = struct{}{}
+	obj := t.expect[k]
+	del := shouldDelete(&obj)
```

Comment:
If we make calling `shouldDelete()` the responsibility of the calling function, this will be simplified

Reply:
Discussing this in [other thread](https://github.com/open-policy-agent/gatekeeper/pull/1014#discussion_r541289197)